library/src/main/java/software/amazon/location/auth/LocationCredentialsProvider.kt

```diff
@@ -248,34 +278,36 @@ class LocationCredentialsProvider {
      * @throws Exception If the Cognito provider is not initialized.
      */
     fun getCredentialsProvider(): aws.sdk.kotlin.services.cognitoidentity.model.Credentials? {
+        val method = securePreferences?.get(METHOD)
+        if (method == "custom" && customCredentials != null) {
+            return customCredentials
+        }
         if (cognitoCredentialsProvider === null) throw Exception("Cognito credentials not initialized")
         return cognitoCredentialsProvider?.getCachedCredentials()
     }
```

Comment:
This and the `createCredentialsProvider` should be rethought considering that the we will be receiving a fully configured `CredentialsProvider`. I think there is unnecessary conversions between `CredentialsProvider` and `aws.sdk.kotlin.services.cognitoidentity.model.Credentials`

Reply:
We are generating credentials with `CognitoIdentityClient` with AWS Kotlin SDK and we are getting this `Credentials(aws.sdk.kotlin.services.cognitoidentity.model)`

and for creating `LocationClient` we need `CredentialsProvider(aws.smithy.kotlin.runtime.auth.awscredentials) `
and for creating `CredentialsProvider` we need `Credentials(aws.smithy.kotlin.runtime.auth.awscredentials)`

So, the conversions are needed in `createCredentialsProvider`

Reply:
Since we `createCredentialsProvider` returns a `CredentialsProvider(aws.smithy.kotlin.runtime.auth.awscredentials)` so why not early return the `this.credentialsProvider` if it is not null (and potentially if `method == "custom"`). Otherwise we are passing a credentials provider to get cerdentials to create a new credentials provider.

Reply:
As discussed internally, can be addressed as fast follow.